src/EventStore.Core/Services/VNode/ClusterVNodeController.cs

```diff
@@ -865,7 +865,16 @@ public class ClusterVNodeController<TStreamId> : ClusterVNodeController, IHandle
 					AdvertiseHost: "",
 					AdvertiseHttpPort: 0,
 					AdvertiseTcpPort: 0);
-
+			if (endpoints.TcpEndPoint != null && (_nodeInfo.ExternalTcp ?? _nodeInfo.ExternalSecureTcp).Address.Equals(IPAddress.Any)) {
+				endpoints = (
```

Comment:
Are you sure you want to overwrite the `endpoints` value incase it's set when there's `_leader` information available?
There's also a null reference warning which I think should addressed.

Reply:
Thank you, I fixed.
I checked the leader is null and `endpoints.TcpEndPoint != null` so  `( _nodeInfo.ExternalTcp ?? _nodeInfo.ExternalSecureTcp)` always != null

Reply:
You mean extract the `GetLeaderInfoEndPoints()` to different class?

Reply:
Don't be afraid to break something, make sure the right tests are in place so you can do the change in a safer way.

The code which is currently added might also break something but you're unaware of it.

To change this, we could extract the method and write tests for the current behavior and add 1 or more tests for the bug you're fixing.

I'm available on Slack if needed.